### PR TITLE
[tinyxml2] Fix generation of pkg-config file

### LINKED ISSUE #10039: tinyxml2 port ships malformed pkg-config .pc on x64-linux
**Describe the bug**
tinyxml2 port ships malformed pkg-config .pc, in particular the file `installed/x64-linux/debug/lib/pkgconfig/tinyxml2.pc` is malformed, as it references to the include directory `/home/straversaro/src/vcpkg/packages/tinyxml2_x64-linux/debug/include` that do not actually contain any include.
~~~
straversaro@iiticublap103:~/src/vcpkg$ cat ./installed/x64-linux/debug/lib/pkgconfig/tinyxml2.pc 
prefix=/home/straversaro/src/vcpkg/packages/tinyxml2_x64-linux/debug
exec_prefix=${prefix}
libdir=${exec_prefix}/lib
includedir=${prefix}/include

Name: TinyXML2
Description: simple, small, C++ XML parser
Version: 7.0.1
Libs: -L${libdir} -ltinyxml2
Cflags: -I${includedir}
~~~

**Environment**
- OS: Ubuntu 18.04 
- Compiler: revision

**To Reproduce**
1. ./vcpkg install tinyxml2
2. See malformed pkg-config file

**Expected behavior**
The pkg-config .pc file should contain valid directories, or not be shipped otherwise.

**Failure logs** 
While configuring the ignition-msgs5 project (see https://bitbucket.org/ignitionrobotics/ign-msgs/src/ignition-msgs5_5.1.0/):
~~~
straversaro@iiticublap103:~/src/vcpkg$ cat  /home/straversaro/src/vcpkg/buildtrees/ignition-msgs5/config-x64-linux-dbg-err.log
CMake Error in src/CMakeLists.txt:
  Imported target "TINYXML2::TINYXML2" includes non-existent path

    "/home/straversaro/src/vcpkg/packages/tinyxml2_x64-linux/debug/include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
~~~

**Additional context**
This problem may be fixed by using the function introduced in https://github.com/microsoft/vcpkg/pull/9861 in the `tinyxml2` port, but I am not sure about this.

### PR DESCRIPTION
**Pull request by orudge**
This change should fix #10039.

### CODE REVIEW

File: ports/tinyxml2/remove-d-suffix.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 67b4f33..4e20f67 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -45,9 +45,6 @@ set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+ set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+ set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
+ 
+-# to distinguish between debug and release lib
+-set(CMAKE_DEBUG_POSTFIX "d")
```
Comment (NancyLi1013) [inline ports/tinyxml2/remove-d-suffix.patch:19]: Why do we remove the suffix 'd' for debug type here? If there is a bug about this, is there another way to fix it?
Generally, we should not rename binaries outside the names given by upstream.
Please refer to this [doc](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#do-not-rename-binaries-outside-the-names-given-by-upstream).

**Review by NancyLi1013 — COMMENTED** [review]
Could you please also bump the version in `CONTROL `file?

**Comment by c72578** [comment]
It should probably be the other way round:
Fix the generated tinyxml2.pc file and add the postfix there in case of a debug build.
Here is a suggestion for an upstream PR:
https://github.com/leethomason/tinyxml2/pull/823

Patch file, tested under vcpkg:
https://patch-diff.githubusercontent.com/raw/leethomason/tinyxml2/pull/823.patch

File: ports/tinyxml2/remove-d-suffix.patch (hunk 1 above)
Comment (orudge) [reply ports/tinyxml2/remove-d-suffix.patch]: I've pushed a new version of the change with @c72578's patch instead. Tested building ignition-msgs5 on macOS and that looks to work.

**Comment by orudge** [comment]
> Could you please also bump the version in `CONTROL `file?

Oops, missed that. Now done, thanks.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by vicroms** [comment]
Thanks for the PR!

**Pull request merged**